src/Features/Core/Portable/RemoveUnusedMembers/AbstractRemoveUnusedMembersDiagnosticAnalyzer.cs

```diff
@@ -14,68 +13,42 @@
-            s_removeUnusedMembersRule = CreateDescriptor(IDEDiagnosticIds.RemoveUnusedMembersDiagnosticId,
-                                                         removeUnusedMembersTitle,
-                                                         removeUnusedMembersMessage,
-                                                         configurable: true,
-                                                         enabledByDefault: true);
-            s_removeUnusedMembersWithFadingRule = CreateUnnecessaryDescriptor(IDEDiagnosticIds.RemoveUnusedMembersDiagnosticId,
-                                                                              removeUnusedMembersTitle,
-                                                                              removeUnusedMembersMessage,
-                                                                              configurable: true,
-                                                                              enabledByDefault: true);
-
-            var removeUnreadMembersTitle = new LocalizableResourceString(nameof(FeaturesResources.Remove_unread_private_members), FeaturesResources.ResourceManager, typeof(FeaturesResources));
-            var removeUnreadMembersMessage = new LocalizableResourceString(nameof(FeaturesResources.Type_0_has_a_private_member_1_that_can_be_removed_as_the_value_assigned_to_it_is_never_read), FeaturesResources.ResourceManager, typeof(FeaturesResources));
-            s_removeUnreadMembersRule = CreateDescriptor(IDEDiagnosticIds.RemoveUnreadMembersDiagnosticId,
-                                                         removeUnreadMembersTitle,
-                                                         removeUnreadMembersMessage,
-                                                         configurable: true,
-                                                         enabledByDefault: true);
-            s_removeUnreadMembersWithFadingRule = CreateUnnecessaryDescriptor(IDEDiagnosticIds.RemoveUnreadMembersDiagnosticId,
-                                                                              removeUnreadMembersTitle,
-                                                                              removeUnreadMembersMessage,
-                                                                              configurable: true,
-                                                                              enabledByDefault: true);
-        }
+        private static DiagnosticDescriptor CreateDescriptor(string id, LocalizableResourceString title, LocalizableResourceString messageFormat)
+            => new DiagnosticDescriptor(id, title, messageFormat, DiagnosticCategory.CodeQuality,
+                    DiagnosticSeverity.Info, isEnabledByDefault: true, customTags: DiagnosticCustomTags.Unnecessary);
```

Comment:
having this be duplicated again saddens me.  the point was to make it so that features didn't have to write all this goop.

Could we extract  base type *below* AbstractCodeStyleDiagnosticAnalzyer.  It could contain everything but the 'options' logic, and this could derive from that and not need to then duplicate all thsi code we already have.